src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/CompilerScope.Storage.cs

```diff
@@ -47,7 +47,9 @@ internal LocalStorage(LambdaCompiler compiler, ParameterExpression variable)
                 // the compiler when emitting an inlined lambda invoke, to
                 // handle ByRef parameters. BlockExpression prevents this
                 // from being exposed to user created trees.
-                _local = compiler.GetNamedLocal(variable.IsByRef ? variable.Type.MakeByRefType() : variable.Type, variable);
+
+                // Set name if DebugInfoGenerator support is brought back.
```

Comment:
This could be done conditionally on whether lambda tree has a DebugInfoGenerator associated with it. Currently it cant have one, since the field is missing, so it seems ok to ignore local name unconditionally.

---
In reply to: [112038242](https://github.com/dotnet/corefx/pull/18564#discussion_r112038242) [](ancestors = 112038242)